### PR TITLE
[Feature]366-Company Details Tabs

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 PR is updated with implemented custom fields. Here is the screenshot:

customFieldsCompanyDetails

### CODE REVIEW

**Comment by turbo124** [comment]
@Civolilah 

This is look good. One small point, on the documents tab, we can remove one level of nesting, so instead of displaying

image

we can show the actual page

image

**Comment by turbo124** [comment]
Forgot to mention, the background color of the TAB bar should match the primary_color of the company.

this property can be found  company?.settings?.primary_color

**Comment by Civolilah** [comment]
> Forgot to mention, the background color of the TAB bar should match the primary_color of the company.
> 
> this property can be found company?.settings?.primary_color

I will replay here for both suggestions with one screenshot. I think that is exactily as you want. Please let me know if it looks good.

companyDetailsNew

**Comment by turbo124** [comment]
@Civolilah this looks good,

@beganovich we should also consider making table headers all primary_color too?

**Comment by Civolilah** [comment]
> @Civolilah this looks good,
> 
> @beganovich we should also consider making table headers all primary_color too?

@turbo124 @beganovich  Okay, thank you for you quick answer. I noticed on the Jira that we have a couple of tasks related to those tabs. Tabs for generate numbers are in progress now, so I will globalize this component on that PR and once we merge it I will apply on this PR and implement for others. While I wait for approval on that PR I will work on different type of tasks from Jira.

**Comment by beganovich** [comment]
I don't think this looks good, few things:

- Tabs are fundamentally wrong here. They're not deep-linked and there's no way to link user to "Company settings > Details" for example?
- I don't think tabs look great, nor they'll look good with primary color applied to them, certainly not with this materialish design applied with deep shadows.
- If we decide to apply primary color to tabs (which I truly hope we don't), all tabs will have to respect this, so base component needs modification.
- We have tab components in the place. Please have a look at Client overview page, there you can see deep linked tabs component which handles active tab, deep linking and all styles you need by wrapping your content in outlet component. There is no need to reinvent. If you need client-sided tabs we have component for them also (in this case you don't).
- For primary color please use `useAccentColor` hook.
- Tables are following accent color, not sure what's happening there.

image

**Comment by turbo124** [comment]
The only issue I have here is there is not a lot of contrast in the tabs, so for a new user, they may not recognize that these are actually Tabs.

@beganovich thoughts?

**Comment by beganovich** [comment]
@turbo124 hm, I mean we do use same component on other pages and it wasn't an issue before? I am open for changes.

**Comment by turbo124** [comment]
Understood, we can iterate on this if needed.

File: src/pages/settings/company/common/hooks/useCompanyDetailsTabs.tsx
Comment (beganovich) [inline src/pages/settings/company/common/hooks/useCompanyDetailsTabs.tsx:19]: This should be the default route. So `href` should be`/settings/company_details`. I believe at the moment it won't highlight the first tab cause of a non-existing route, right?

File: src/pages/settings/routes.tsx
Comment (beganovich) [inline src/pages/settings/routes.tsx:27]: @turbo124 do we need custom fields tab here? It wasn't there before.

**Review by beganovich — CHANGES_REQUESTED** [review]
Looks great. Minor changes needed :+1:

File: src/pages/settings/company/common/hooks/useCompanyDetailsTabs.tsx
Comment (Civolilah) [reply src/pages/settings/company/common/hooks/useCompanyDetailsTabs.tsx]: > 

Yeah sure, I missed that, now it's changed.

File: src/pages/settings/routes.tsx
Comment (Civolilah) [reply src/pages/settings/routes.tsx:26]: > @turbo124 do we need custom fields tab here? It wasn't there before.

David mentioned me on the Jira that he wants to add this tab same as on the Flutter app. So, I just prepared everything for doing this piece of task. I will ping him to give me more instructions for this tab.

Comment (turbo124) [reply src/pages/settings/routes.tsx:26]: @beganovich I can't see where in the app we show company custom fields currently, so i advised @Civolilah to put them here

**Pull request merged**
